### PR TITLE
Product Component

### PR DESCRIPTION
**Pull request by dandelionadia**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -30,6 +38,23 @@ const App: React.FC = () => {
...
       <header>
         <a href="https://reactjs.org">Learn React</a>
+        <Box flex paddingHorizontal={3.4}>
+          <Product
+            image="
+            https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/1-21-620x698.jpg"
+            secondImage="https://demo2.wpopal.com/auros/wp-content/uploads/2018/10/2-21-600x675.jpg"
+            productName="Mega 2 Seater Sofa"
+            price={160.71}
+            buttonText=" add to card"
```
Comment (kettanaito) [inline src/App.tsx]: Don't add the space here. If you need the space, add it before the value of `{buttonText}` where you use it (in `Product.jsx`). 

```diff
- buttonText=" add to card"
+ buttonText="add to card" // in App
+ <p>Some {buttonText} spacing</p> // in Product
```

File: src/atoms/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  background-color: ${({ theme }) => theme.colors.black};
+  color: ${({ theme }) => theme.colors.white};
+  font-weight: 400;
+  font-size: 16px;
+  padding: 16px;
+  margin: 15px;
+  border: 0;
+  width: 100%;
+
+  hover {
```
Comment (kettanaito) [inline src/atoms/Button.tsx]: I think this should be `:hover`.

File: src/atoms/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    background-color: ${({ theme }) => theme.colors.orangeLight};
+  }
+`
+
+const ProductButton = styled(Button)`
```
Comment (kettanaito) [inline src/atoms/Button.tsx]: `ProductButton` is too specific to be placed in the `Button.jsx`. Instead, move this `ProductButton` declaration to `Product.jsx`.

File: src/atoms/Text.tsx [hunk 4]
```diff
@@ -13,6 +13,8 @@ const Text = styled.p<TextProps>`
     css`
       color: ${({ theme }) => theme.colors.blackLight};
       font-size: 16px;
+      font-weight: 600;
```
Comment (kettanaito) [inline src/atoms/Text.tsx]: This makes all paragraphs on the page bold (`font-weight: 600`). Are you sure that's intended?

File: src/molecules/Product.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      onMouseEnter={() => setIsHover(true)}
+      onMouseLeave={() => setIsHover(false)}
+    >
+      <Box as={StyledContainerImage}>
+        <a href="#">
+          <img src={isHover ? secondImage : image} alt="img" />
+        </a>
+        {isHover && (
+          <ProductButton>
+            <p>+ {buttonText}</p>
```
Comment (kettanaito) [inline src/molecules/Product.tsx]: This usage of spacing is absolutely right! You can remove the space in `App.jsx` from your `buttonText="..."` props.

File: src/theme.ts [hunk 6]
```diff
@@ -6,6 +6,7 @@ export default {
     black: '#000',
     blackLight: '#222',
     orange: '#eb7025',
-    orangeLight: '#ffb14b'
+    orangeLight: '#ffb14b',
+    white: 'white'
```
Comment (kettanaito) [inline src/theme.ts]: Nitpick, but `#fff` is shorter than `white` :P

File: src/molecules/Product.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  max-width: 500px;
+  height: auto;
+`
+const StyledProductDescription = styled.div`
+  text-align: center;
+  padding: 25px 0;
+`
+const StyledContainerImage = styled.div`
+  position: relative;
+  z-index: 0;
```
Comment (kettanaito) [inline src/molecules/Product.tsx]: I don't think this does something. Try removing `z-index` and see if it changes something. If not, `z-index` is not needed.

File: src/molecules/Product.tsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import { Text } from '../atoms/Text'
+import { ProductButton } from '../atoms/Button'
+
+const StyledContainerProduct = styled.div`
+  max-width: 500px;
+  height: auto;
+`
+const StyledProductDescription = styled.div`
+  text-align: center;
+  padding: 25px 0;
```
Comment (kettanaito) [inline src/molecules/Product.tsx]: Well, technically, if you wish to split spacing and other visual styling you should move `padding: 25px 0` from here to the `<Box as={StyledProductDescription}>`:

```jsx
<Box as={StyledProductDescription} paddingVertical={25 / 16}>
```

P.S. This is not a serious mistake, feel free to ignore.

File: src/atoms/Text.tsx (hunk 4 above)
Comment (dandelionadia) [reply src/atoms/Text.tsx]: it is not for all paragraphs, just with props 'big'
 ${props =>
    props.big &&
    css`
      color: ${({ theme }) => theme.colors.blackLight};
      font-size: 16px;
      font-weight: 600;
      margin: 0;
    `}

File: src/theme.ts (hunk 6 above)
Comment (dandelionadia) [reply src/theme.ts]: It is okay. I appreciate this :)

File: src/molecules/Product.tsx (hunk 8 above)
Comment (dandelionadia) [reply src/molecules/Product.tsx]: I like it more.

**Pull request merged**